sdv/single_table/base.py

```diff
@@ -234,3 +236,50 @@ def validate(self, data):
 
         if errors:
             raise InvalidDataError(errors)
+
+    def _validate_transformers(self, column_name_to_transformer):
+        keys = self._get_primary_and_alternate_keys() | self._get_set_of_sequence_keys()
+        for column, transformer in column_name_to_transformer.items():
+            transformer_name = type(transformer).__name__
+            sdtype = self.metadata._columns[column]['sdtype']
+            if sdtype != transformer.get_input_sdtype():
+                raise Error(
+                    f"Column '{column}' is a {sdtype} column, which is incompatible "
+                    f"with the '{transformer_name}' transformer."
+                )
+
+            if column in keys and transformer_name not in {'AnonymizedFaker', 'RegexGenerator'}:
+                raise Error(
+                    f"Column '{column}' is a key. It cannot be preprocessed using "
+                    f"the '{transformer_name}' transformer."
+                )
+
+            # If columns were set, the transformer was fitted
+            if transformer.columns:
+                raise Error(f"Transformer for column '{column}' has already been fit on data.")
+
+    def _warn(self, column_name_to_transformer):
```

Comment:
minor: maybe make this more descriptive. `_raise_transformer_update_warnings` or `_warn_for_transformer_updates`